stage3.Dockerfile

```diff
@@ -7,24 +7,27 @@
 # to fetch the signing key.
 
 ARG BOOTSTRAP
-FROM ${BOOTSTRAP:-alpine:3.6} as builder
+FROM ${BOOTSTRAP:-alpine:3.7} as builder
 
 WORKDIR /gentoo
 
 ARG ARCH=amd64
 ARG MICROARCH=amd64
 ARG SUFFIX
-ARG DIST="http://distfiles.gentoo.org/releases/${ARCH}/autobuilds/"
+ARG DIST="https://ftp-osl.osuosl.org/pub/gentoo/releases/${ARCH}/autobuilds"
 ARG SIGNING_KEY="0xBB572E0E2D182910"
 
 RUN echo "Building Gentoo Container image for ${ARCH} ${SUFFIX} fetching from ${DIST}" \
  && apk --no-cache add gnupg tar wget \
- && STAGE3PATH="$(wget -q -O- "${DIST}/latest-stage3-${MICROARCH}${SUFFIX}.txt" | tail -n 1 | cut -f 1 -d ' ')" \
+ && STAGE3PATH="$(wget -O- "${DIST}/latest-stage3-${MICROARCH}${SUFFIX}.txt" | tail -n 1 | cut -f 1 -d ' ')" \
+ && echo "STAGE3PATH:" $STAGE3PATH \
  && STAGE3="$(basename ${STAGE3PATH})" \
- && wget -q -c "${DIST}/${STAGE3PATH}" "${DIST}/${STAGE3PATH}.CONTENTS" "${DIST}/${STAGE3PATH}.DIGESTS.asc" \
- && gpg --keyserver hkps.pool.sks-keyservers.net --recv-keys ${SIGNING_KEY} \
- || gpg --keyserver keys.gnupg.net --recv-keys ${SIGNING_KEY} \
- || gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys ${SIGNING_KEY} \
+ && wget -q "${DIST}/${STAGE3PATH}" "${DIST}/${STAGE3PATH}.CONTENTS" "${DIST}/${STAGE3PATH}.DIGESTS.asc" \
+ && gpg --list-keys \
```

Comment:
Is the list keys here necessary or just debug?

Reply:
It is necessary - it forces pre-creation of `/root/.gnupg` and contents with correct permissions, so the following config-updates find the files already present without worrying about permissions.